collect_performance.py

```diff
@@ -1,118 +1,205 @@
 from pathlib import Path
-import pyarrow.parquet as pq
 import os
-import yaml
 import pandas as pd
 import argparse
 
-def setup_eval_df(eval_file: str|os.PathLike, team: str, identifier: str, target: str, window: str) -> pd.DataFrame:
-    df: pd.DataFrame = pq.read_table(eval_file).to_pandas()
-    df["window"] = window
-    df["target"] = target
-    df["team"] = team
-    df["identifier"] = identifier
-    unit: str = df.index.name
-    df.reset_index(inplace = True)
-    df.set_index(["team", "identifier", "target", unit, "window"], inplace = True)
-    return df
+from utilities import list_submissions, get_target_data, views_month_id_to_year, views_month_id_to_month, TargetType, get_submission_details, data_in_target
+
+
+def get_eval(submission: str|os.PathLike, target: TargetType, groupby: str|list[str] = None, across_submissions: bool = False) -> pd.DataFrame:
+    """Convenience function to read and aggregate evaluation metrics from a submission.
+
+    Parameters
+    ----------
+    submission : str | os.PathLike
+        Path to a folder structured like a submission_template
+    target : TargetType
+        A string, either "pgm" for PRIO-GRID-months, or "cm" for country-months.
+    groupby : str | list[str], optional
+        A dimension to aggregate results across. Some options (all except None and "pooled" can be combined in a list):
+        None: no aggregation
+        "pooled": complete aggregation
+        "year": aggregate by calendar year
+        "month": aggregate by calendar month
+        "month_id": aggregate by month_id (1 is January 1980)
+        "country_id": aggregate by country (currently only works for target == "cm")
+        "priogrid_gid": aggregate by PRIO-GRID id.
+    across_submissions : bool
+        Aggregate across submissions. Default false (i.e., aggregate by [team, model])
+
+    Returns
+    -------
+    pandas.DataFrame
+
+    Raises
+    ------
+    ValueError
+        Target must be "cm" or "pgm".
+    FileNotFoundError
+        There must be .parquet-files in the submission/{target} sub-folders.
+    """
+
+    if target == "cm":
+        unit = "country_id"
+    elif target == "pgm":
+        unit = "priogrid_gid"
+    else:
+        raise ValueError(f'Target must be either "cm" or "pgm".')
+
+    submission = Path(submission)
+    if not data_in_target(submission, target):
+        raise FileNotFoundError
+
+    groupby_inner = groupby.copy()
```

Comment:
I think I might be able to do something like `if groupby == None: groupby = []`instead. It is written like it is due to the [mutable default argument](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument) behavior in Python. If I simply use groupby, it is not instantiated many times with the default argument, rather, it will reuse the function-parameter from the last call. It was not fun to debug, but apparently "Python’s handling of default parameter values is one of a few things that tends to trip up most new Python programmers (but usually only once)." [source](https://web.archive.org/web/20200221224620id_/http://effbot.org/zone/default-values.htm)  So, I was not alone on that one.